assets/src/scripts/components/hydrograph/graph-controls.spec.js

```diff
@@ -0,0 +1,243 @@
+        component.append('div').attr('class', 'provisional-data-alert');
+
+        graphNode = document.getElementById('hydrograph');
+    });
+
+    afterEach(() => {
+        select('#hydrograph').remove();
+    });
+
+    describe('graphControls', () => {
+
+        let store;
+        beforeEach(() => {
+            store = configureStore(TEST_STATE);
+            attachToNode(store, graphNode, {siteno: '12345678'});
+        });
+
+        // last year checkbox tests
+        it('Should render the compare toggle checked', () => {
+            const checkbox = select('#last-year-checkbox');
+            expect(checkbox.size()).toBe(1);
+            expect(checkbox.property('checked')).toBe(true);
+        });
+
+        it('Should render the compare toggle unchecked', () => {
+            store.dispatch(Actions.toggleTimeSeries('compare', false));
+            const checkbox = select('#last-year-checkbox');
+            expect(checkbox.size()).toBe(1);
+            expect(checkbox.property('checked')).toBe(false);
+        });
+
+        it('should be enabled if there are last year data', () => {
+            expect(select('#last-year-checkbox').property('disabled')).toBeFalsy();
+        });
+
+        it('should be disabled if there are no last year data', () => {
+            store.dispatch(Actions.setCurrentVariable('45807190'));
+            expect(select('#last-year-checkbox').property('disabled')).toBeTruthy();
+        });
+
+        it('Should render one lines', () => {
+            expect(selectAll('#ts-compare-group .line-segment').size()).toBe(1);
+        });
+
+        it('Should remove the lines when removing the compare time series', () => {
+            store.dispatch(Actions.toggleTimeSeries('compare', false));
+            expect(selectAll('#ts-compare-group .line-segment').size()).toBe(0);
+        });
+
+        // median checkbox tests
+        it('Should render the median toggle checked', () => {
+            const checkbox = select('#median-checkbox');
+            expect(checkbox.size()).toBe(1);
+            expect(checkbox.property('checked')).toBe(true);
+        });
+
+        it('Should render the median toggle unchecked', () => {
+            store.dispatch(Actions.toggleTimeSeries('median', false));
+            const checkbox = select('#median-checkbox');
+            expect(checkbox.size()).toBe(1);
+            expect(checkbox.property('checked')).toBe(false);
+        });
+
+        it('should be enabled if there are median statistics data', () => {
+            expect(select('#median-checkbox').property('disabled')).toBeFalsy();
+        });
+
+        it('should be disabled if there are no median statistics data', () => {
+            store.dispatch(Actions.setCurrentVariable('45807190'));
+            expect(select('#median-checkbox').property('disabled')).toBeTruthy();
+        });
+
+        it('Should render one lines', () => {
```

Comment:
Same as comment on line 202 except this can be added to index.js.

Reply:
done!